cipher/src/block.rs

```diff
@@ -127,19 +127,37 @@ pub trait BlockCipherEncrypt: BlockSizeUser + Sized {
         self.encrypt_padded_inout::<P>(buf)
     }
 
-    /// Pad input and encrypt into a newly allocated Vec. Returns resulting ciphertext Vec.
+    /// Pad `msg` with padding algorithm `P`, encrypt it into a newly allocated `Vec`,
+    /// and return the resulting ciphertext vector.
+    ///
+    /// # Panics
+    /// If `NoPadding` is used with a message size that is not a multiple of the cipher block size.
     #[cfg(all(feature = "block-padding", feature = "alloc"))]
     #[inline]
-    fn encrypt_padded_vec<P: Padding>(&self, msg: &[u8]) -> Vec<u8> {
+    fn encrypt_padded_vec<P: Padding + 'static>(&self, msg: &[u8]) -> Vec<u8> {
```

Comment:
I probably will release `block-padding` v0.4.2 with added `'static` bound on `Padding` and will yank v0.4.1.